.github/workflows/test.yml

```diff
@@ -0,0 +1,21 @@
+name: test
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches: [ main ]
+
+permissions:
+  contents: write
```

Comment:
why do we need write permissions to run the tests?

Reply:
No idea. I copy the test.yml from another project. I will remove and check if it works.

Reply:
Deleted and still works - fixed